app/internal/handlers/handler.go

```diff
@@ -0,0 +1,365 @@
+		http.Error(w, `{"error":"Ошибка формирования JSON-ответа"}`, http.StatusInternalServerError)
+		return
+	}
+}
+
+// получение одной задачи по ID
+func GetOneTask(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, `{"error":"Метод не поддерживается"}`, http.StatusMethodNotAllowed)
+		return
+	}
+	idStr := chi.URLParam(r, "id")
+	if idStr == "" {
+		idStr = r.URL.Query().Get("id")
+	}
+
+	id, err := strconv.Atoi(idStr)
+	if err != nil {
+		http.Error(w, `{"error":"Некорректный идентификатор"}`, http.StatusBadRequest)
+		return
+	}
+	// получаем задачу
+	task, err := commandsDB.GetTaskByID(id)
+	if err != nil {
+		http.Error(w, fmt.Sprintf(`{"error":"%s"}`, err.Error()), http.StatusNotFound)
+		return
+	}
+
+	response := map[string]string{
+		"id":      task.ID,
+		"date":    task.Date,
+		"title":   task.Title,
+		"comment": task.Comment,
+		"repeat":  task.Repeat,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		http.Error(w, `{"error":"Ошибка формирования JSON-ответа"}`, http.StatusInternalServerError)
+		return
+	}
+}
+
+// обновление задачи
+func PutTask(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPut {
+		http.Error(w, `{"error":"Метод не поддерживается"}`, http.StatusMethodNotAllowed)
+		return
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	var task models.Remind
+	if err := decoder.Decode(&task); err != nil {
+		http.Error(w, `{"error":"Ошибка декодирования JSON"}`, http.StatusBadRequest)
+		return
+	}
+
+	// проверяем наличие поля id
+	if task.ID == "" {
+		http.Error(w, `{"error":"Не указан идентификатор задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	// проверяем также поле title
+	if task.Title == "" {
+		http.Error(w, `{"error":"Не указан заголовок задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	// проверяем дату
+	now := time.Now()
+	today := now.Format("20060102")
+	if task.Date == "" || task.Date == "today" {
+		task.Date = today
+	} else {
+		parsedDate, err := time.Parse("20060102", task.Date)
+		if err != nil {
+			http.Error(w, `{"error":"Некорректный формат даты"}`, http.StatusBadRequest)
+			return
+		}
+		if parsedDate.Before(time.Now()) {
+			if task.Repeat == "" {
+				task.Repeat = today
+			} else {
+				nextDate, err := tasksRules.NextDate(now, task.Date, task.Repeat)
+				if err != nil {
+					http.Error(w, `{"error":"Некорректное правило повторения"}`, http.StatusBadRequest)
+					return
+				}
+				if task.Date != today {
+					task.Date = nextDate
+				}
+			}
+		}
+	}
+	// проверяем правило повторения
+	if task.Repeat != "" {
+		_, err := tasksRules.NextDate(now, task.Date, task.Repeat)
+		if err != nil {
+			http.Error(w, `{"error":"Некорректное правило повторения"}`, http.StatusBadRequest)
+			return
+		}
+	}
+
+	// Обновляем задачу в базе данных
+	err := commandsDB.UpdateTask(task)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Лучше так сравнивать, если ошибку обернуть, то такое сравнение не сработает уже
```suggestion
		if errors.Is(err, sql.ErrNoRows) {
```